tests/testbed_setup/test_xr_migration.py

```diff
@@ -0,0 +1,210 @@
+            if found:
+                break
+        except Exception:
+            continue
+    return output
+
+
+def test_xr_migration(duthost_console, duthosts, enum_supervisor_dut_hostname, creds):
+    duthost = duthosts[enum_supervisor_dut_hostname]
+    dut_hostname = duthost.hostname
+
+    # gather console information
+    duthost_console.timeout = 300
+    console_user = creds['console_user']['console_ssh']
+    console_password = creds['console_password']['console_ssh'][0]
+    prompt = duthost_console.find_prompt()[:-1]
+
+    # gather minigraph files from DUT for config restore
+    duthost_console.write_channel("show chassis modules midplane-status" + duthost_console.RETURN +
+                                  duthost_console.RETURN)
+    split = read_con(duthost_console, prompt).split("\n")
+    for line in split[3:-2]:
+        split_line = line.split(" ")
+        if split_line[len(split_line) - 1].strip() == "True":
+            card_index = split_line[1][-1]
+            logger.debug(f"found card: {split_line[1]} with index: {card_index}")
+            logger.debug(f"split line: {split_line}")
+            logger.debug(f"ip is: {split_line[8]}")
+            filename = "minigraph-" + dut_hostname + "-lc0" + split_line[1][-1] + ".xml"
+            logger.debug(f"filename: {filename}")
+            duthost_console.write_and_poll(f"scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null \
+                                           {console_user}@{split_line[8]}:/etc/sonic/minigraph.xml \
+                                           {scp_user}@{scp_host}:{mig_file_path}{filename}" + duthost_console.RETURN,
+                                           "password:")
+            duthost_console.write_and_poll(scp_pass + duthost_console.RETURN, "password:")
+            duthost_console.write_channel(console_password + duthost_console.RETURN)
+            read_con(duthost_console, prompt)
+
+    # copy RP minigraph
+    filename = "minigraph-" + dut_hostname + "-sup00.xml"
+    logger.debug(f"filename: {filename}")
+    duthost_console.write_and_poll(f"scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null \
+                                    /etc/sonic/minigraph.xml {scp_user}@{scp_host}:{mig_file_path}{filename}" +
+                                   duthost_console.RETURN, "password:")
+    # duthost_console.write_and_poll(scp_pass + duthost_console.RETURN, "password:")
+    duthost_console.write_channel(scp_pass + duthost_console.RETURN)
+    read_con(duthost_console, prompt)
+
+    # copy all files from rollback-files directory to duthost
+    duthost_console.write_and_poll(f"sudo scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null \
+                                   {scp_user}@{scp_host}:{rollback_file_path}* /host/", "password:")
+    duthost_console.write_channel(scp_pass + duthost_console.RETURN)
+    read_con(duthost_console, prompt)
+
+    # Migrate to XR Rollback Image
```

Comment:
Could you point out where and how do we provide the image file?

Reply:
The images need to be in the migration and rollback directories that are declared at the top of the file.

Reply:
Need a logic to assert the necessary files are prepared already,
If the preparation is not ready, we should actively stop the test and raise the decent error message.
Rather than waiting the test to run and passively stop.

Reply:
The files have to be staged somewhere the DUT has scp access to.  I am not aware of a way to use scp to verify the files in a remote directory.  This test is not able to be setup as usual sonic-mgmt tests, it needs a lot of manual setup and file staging.  I am open to suggestion as to how to check a scp directory for files, but I am of the opinion that if you are running this test case you know what needs to be staged and where and have changed the variables in the test to match what you setup.

Reply:
I have added directions at the top of the test to specify the variables that need to be updated and what files need to be staged.